app/view/lineSliceGridExample/LineSliceGrid.js

```diff
@@ -0,0 +1,113 @@
+var store = Ext.create('Ext.data.Store', {
+    data: [
+        {startMeasure: 0, endMeasure: 200, text: 'slice 1'},
+        {startMeasure: 200, endMeasure: 1000, text: 'slice 2'},
+        {startMeasure: 1100, endMeasure: 1800, text: 'slice 3'},
+        {startMeasure: 1400, endMeasure: 4000, text: 'slice 4'},
+        {startMeasure: 5000, endMeasure: 12000, text: 'slice 5'}
+    ]
+});
+
+function meterRenderer(val) {
+    return val.toFixed(1) + ' m';
+}
+
+Ext.define('CpsiMapview.view.lineSliceGridExample.LineSliceGrid', {
+    extend: 'Ext.grid.Panel',
+    xtype: 'cmv_line_slice_grid',
+
+    requires: [
+        'CpsiMapview.plugin.LineSliceHighlight'
+    ],
+
+    plugins: [{
+        ptype: 'cmv_line_slice_highlight'
+    }],
+
+    store: store,
+
+    columns: [
+        {
+            text: 'From',
+            dataIndex: 'startMeasure',
+            renderer: meterRenderer
+        },
+        {
+            text: 'To',
+            dataIndex: 'endMeasure',
+            renderer: meterRenderer
+        },
+        {
+            text: 'Text',
+            dataIndex: 'text'
+        }
+    ],
+    listeners: {
+        itemclick: function (_, record) {
```

Comment:
Wouldn't it make sense that the plugin registers the event itself, so that by using the plugin the functionality is automatically available in the father grid?

Reply:
Yes that would be possible, but would only work if the properties in the used model are named exactly `startMeasure` and `endMeasure`. We could test if that is the case and if someone would want to use other properties he or she could still call the `highlightSlice` method manually.

Reply:
Also the plugin does not have access to the geometry as it is. At the moment the method `highlightSlice` accepts a geometry as a parameter. Instead we could have a `setGeometry` method - then this could be done.

Reply:
OK understand, just leave it as it is. Even more flexible the way it is done now.